Assets/Scripts/Interactables/Dialogue/SpeakerUI.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine.UI;
+using UnityEngine;
+using TMPro;
+
+public class SpeakerUI : MonoBehaviour
+{
+    public TMP_Text dialogue;
+    public GameObject textBoxContainer;
+    
+    private Vector3 textPosition;
+
+    void Start()
+    {
+        this.Hide();
+    }
+
+    public string Dialogue
+    {
+        set { dialogue.text = value;}
+    }
+
+    public void Update()
+    {
+
+        textPosition = Camera.main.WorldToScreenPoint(this.transform.position);

```

Comment:
You can just do `transform.position` (the `this.` is always implied).

(Also can we remove the empty lines around these two lines of code? 😝)